docs/architecture/index.md

```diff
@@ -1,86 +1,32 @@
-# System Design Note
+# Overview
 
-This design document contains notes that are relevant to the MXNet system design and deep learning
-libraries in general. We believe that open sourcing this system design note can help general audiences understand the  motivations, benefits and drawbacks of our design choices. This will help deep learning practitioners as well as builders of other deep learning systems.
+This design document contains notes that are relevant to the MXNet system design and deep learning libraries in general. We believe that open sourcing this system design note can help general audiences understand the  motivations, benefits and drawbacks of our design choices. This will help deep learning practitioners as well as builders of other deep learning systems.
 
-## Deep Learning Design Notes
+# Deep Learning System Design Concepts
 
-This section will be updated with self-contained design notes on various aspect of deep learning systems,
-in terms of abstraction, optimization and trade-off.
+In this section, we will describe general design concepts of Deep Learning systems. We will mainly focus on following 3 areas influencing the design and architecture of a Deep Learning System.
+1. Abstraction
+2. Optimization
+3. Trade offs
 
-* [Programming Models for Deep Learning](program_model.md)
-* [Dependency Engine for Deep Learning](note_engine.md)
-* [Squeeze the Memory Consumption of Deep Learning](note_memory.md)
-* [Efficient Data Loading Module for Deep Learning](note_data_loading.md)
-* [Survey of RNN Interface](rnn_interface.md)
 
-The next parts will be specific to MXNet
+&nbsp;
 
-## Overview of the Design
+Refer below sections for more details
+* [Programming Models for Deep Learning](http://mxnet.io/architecture/program_model.html)
```

Comment:
Yes this is important. I will take up in next iteration. 
With this change, I am bringing all related contents together and adding "recommended next steps" to guide users logically next topics.